metadata-ingestion/src/datahub/ingestion/source/affirm/artifact.py

```diff
@@ -99,47 +75,40 @@ def create(cls, config_dict, ctx):
     def get_workunits(self) -> Iterable[MetadataWorkUnit]:
         directory = self.config.directory
         platform = self.config.platform
+        platform_instance = self.config.platform_instance
         env = self.config.env
         for artifact in iterate_artifact(directory):
-            dataset_name = artifact.name
-            logging.info(f'> Processing dataset {dataset_name}')
-            dataset_urn = builder.make_dataset_urn(platform, dataset_name, env)
-            dataset_snapshot = DatasetSnapshot(
-                urn=dataset_urn,
-                aspects=[],
+            dataset_name = (
+                f'{artifact.schema_name}.{artifact.name}'
+                if len(artifact.schema_name) > 0
+                else artifact.name
             )
-            # set up dataset properties 
-            dataset_properties = DatasetPropertiesClass(
-                description=artifact.description,
-                tags=[],
-                customProperties={
-                    'privacy_entrypoint': artifact.privacy_entrypoint,
-                    'retention_days': f'{artifact.retention_days}'
-                }
+            logger.info(f'Processing dataset {dataset_name}')
+            dataset_urn = builder.make_dataset_urn_with_platform_instance(
+                platform=platform,
+                name=dataset_name,
+                platform_instance=platform_instance,
+                env=env
             )
-            dataset_snapshot.aspects.append(dataset_properties)
-            # set up ownership
-            ownership = OwnershipClass(
-                owners=[
-                    OwnerClass(
-                        owner= builder.make_group_urn(make_groupname(artifact.owner)),
-                        type=OwnershipTypeClass.DATAOWNER
-                    )
-                ]
+
+            mcp = MetadataChangeProposalWrapper(
```

Comment:
@amit-ramesh changed to use `MCP` now. tested and found it is not fully overwriting the existing datahub datasets yay